Update positionally.js

theRealDadsdy opened the pull request:
Fixed minus, p, EOF

theRealDadsdy committed:
Update positionally.js

Fixed minus, p, EOF

chunkybanana commented on positionally.js:
```diff
@@ -145,7 +146,7 @@ let positionally = (code, inputs = [], flags = '', output = x => process.stdout.
                     } else if (char == 'd') {
                         let [a, b] = pop(2);
-                        push(b / a | 0);
+                        push(b / a || 0);
```
This is to make it floor division (a | 0 = Math.floor(a)).

chunkybanana commented on positionally.js:
```diff
@@ -5,6 +5,7 @@ let positionally = (code, inputs = [], flags = '', output = x => process.stdout.
     code = code.map(x => x.padEnd(maxlength, ' ').split``);
 
     inputs = inputs.flatMap(x => typeof x == 'string' ? x.split``.map(x => x.charCodeAt()) : x);
+    inputs.push(-1)
```
Probably want to append -1 to each string rather than to the end of everything

theRealDadsdy replied:
Shouldn't there still be a -1 at the end so it doesn't err on no input?

theRealDadsdy committed:
Update positionally.js

Fixed s command

theRealDadsdy committed:
Update README.md

Slightly more understandable

theRealDadsdy committed:
Update README.md

More readable

theRealDadsdy committed:
Update positionally.js

theRealDadsdy committed:
0 is now eof

theRealDadsdy committed:
EOF is 0

chunkybanana commented on positionally.js:
```diff
@@ -4,7 +4,8 @@ let positionally = (code, inputs = [], flags = '', output = x => process.stdout.
 
-    inputs = inputs.flatMap(x => typeof x == 'string' ? x.split``.map(x => x.charCodeAt()) : x);
+    inputs = inputs.flatMap(x => typeof x == 'string' ? x.split``.map(x => x.charCodeAt()).concat(0) : x);
+    inputs.push(0)
```
I still don't think there should be a 0 at the end of all inputs. Besides, how can this be distinguished from a zero-length string?

theRealDadsdy commented:
On Sat, Jul 22, 2023 at 19:21 Chunkybanana ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In positionally.js
> <https://github.com/chunkybanana/positionally/pull/3#discussion_r1271358207>
> :
>
> > @@ -4,7 +4,8 @@ let positionally = (code, inputs = [], flags = '', output = x => process.stdout.
>      let maxlength = Math.max(...code.map(x => x.length));
>      code = code.map(x => x.padEnd(maxlength, ' ').split``);
>
> -    inputs = inputs.flatMap(x => typeof x == 'string' ? x.split``.map(x => x.charCodeAt()) : x);
> +    inputs = inputs.flatMap(x => typeof x == 'string' ? x.split``.map(x => x.charCodeAt()).concat(0) : x);
> +    inputs.push(0)
>
> I still don't think there should be a 0 at the end of all inputs. Besides,
> how can this be distinguished from a zero-length string?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chunkybanana/positionally/pull/3#pullrequestreview-1542160479>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWTUD5STRQQJIXC2MY6VBGTXRRN6ZANCNFSM6AAAAAA2TT5M54>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
It wouldn’t be able to be distinguished from a zero-length string, but it’s
still better than not having an EOF. Anything you can do without an EOF you
can do with an EOF. However, the converse isn’t true.

theRealDadsdy committed:
Removed ending from input unless it’s the only input
